azure-iot-device/azure/iot/device/common/mqtt_provider.py

```diff
@@ -120,23 +120,40 @@ def on_message(client, userdata, mqtt_message):
 
         logger.info("Created MQTT provider, assigned callbacks")
 
-    def connect(self, password):
+    def _create_ssl_context(self, x509=None):
         """
-        Connect to the MQTT broker, using hostname and username set at instantiation.
-
-        This method should be called as an entry point before sending any telemetry.
-
-        :param str password: The password for connecting with the MQTT broker.
+        This method creates the SSLContext object used by Paho to authenticate the connection.
+        :param x509: The x509 certificate. If present then the SSLContext is created using x509.
         """
-        logger.info("connecting to mqtt broker")
-
+        logger.info("creating a SSL context")
         ssl_context = ssl.SSLContext(protocol=ssl.PROTOCOL_TLSv1_2)
+
         if self._ca_cert:
             ssl_context.load_verify_locations(cadata=self._ca_cert)
         else:
             ssl_context.load_default_certs()
         ssl_context.verify_mode = ssl.CERT_REQUIRED
         ssl_context.check_hostname = True
+
+        if x509 is not None:
+            logger.info("configuring SSL context with client-side certificate and key")
+            ssl_context.load_cert_chain(x509._cert, x509._key, x509._pass_phrase)
```

Comment:
>ssl_context.load_cert_chain(x509._cert, x509._key, x509._pass_phrase) [](start = 12, length = 69)

Should there be an X509 model in common/models?  I think it needs to be somewhere and I think this is the right location since this is shared between provisioning and iothub.

Reply:
I agree, it should be in common.

---
In reply to: [291675754](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r291675754) [](ancestors = 291675754)

Reply:
yes after creating the class I too believe that the  X509 class should be shifted to common. But there is no models in common. I would create one. 

---
In reply to: [291675754](https://github.com/Azure/azure-iot-sdk-python-preview/pull/105#discussion_r291675754) [](ancestors = 291675754)